src/modules/postcode-lookup/index.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * Returns a postocde lookup module.
+ * @param {Object} window  Global window object
+ */
+function createPostcodeLookup(window) {
+    // Define an empty JSON object to temporarily store matched address results.
+    // eslint-disable-next-line no-unused-vars
+    let tmpAddressSearchResultsJson = {};
+
+    function addSearchResultsToSelectElement(data) {
+        const selectElementResults = window.document.getElementById(
+            'address-search-results-dropdown'
+        );
+        // Clear the results drop-down list.
+        const options = window.document.querySelectorAll('#address-search-results-dropdown option');
```

Comment:
[getElementsByTagName](https://developer.mozilla.org/en-US/docs/Web/API/Element/getElementsByTagName) returns a Live HTMLCollection so you could do something like:

`selectElementResults.getElementsByTagName('options');` and this list would be updated on the fly automatically. So you could move this to outside this function also and you would not need to activelly query the DOM again.